addons/web_editor/static/src/js/wysiwyg/widgets/media.js

```diff
@@ -1060,7 +1060,7 @@ var DocumentWidget = FileWidget.extend({
     _getAttachmentsDomain: function (needle) {
         var domain = this._super.apply(this, arguments);
         // the assets should not be part of the documents
-        return domain.concat('!', utils.assetsDomain());
+        return ['&', '|', ['url', '=', null], '!', ['url', '=like', '/%/assets/%']].concat(domain);
```

Comment:
Maybe improve the comment line to reflect the python method name you are mimicking here. (Following https://github.com/odoo/odoo/pull/89151#discussion_r863105558)
Also, do we need to first %s? Isn't it always /web/assets? (You can leave it like that, just to know if you found something else during your investigation)
Either way that's not wrong, at worst it is a bit too loose but that's not even something bad.

After that we can ship it, sorry to be picky :grimacing: